tracks/sousprogrammes/sp1-01-intro/mission.md

```diff
@@ -0,0 +1,71 @@
+### Notions théoriques
+
+Comme dans les langages de haut-niveau, il est possible d'écrire des sous-programmes en Pep/8, la convention d'appel en revanche change un peu des langages de haut niveau.
+
+En Pep/8, deux instructions sont disponibles pour travailler avec des sous-programmes : `CALL` et `RET0`.
+
+`CALL` comme son nom l'indique, sert à appeler un sous-programme. Il fonctionne d'une façon analogue à `BR`, on lui donne un symbole ou une adresse, il va effectuer le branchement et garder un moyen de revenir au flux d'exécution avant l'appel.
+
+On discutera plus en détail de quoi et comment les informations sont gardées ultérieurement.
+
+`RET0` quant à lui sert lorque la fonction a fini d'éxécuter à retourner au point d'appel. Il est donc essentiel de l'appeler à la terminaison de la fonction pour revenir au lieu d'appel et continuer normalement le programme.
+
+En d'autres mots, l'instruction `CALL` s'utilise comme un branchement inconditionnel (`BR`) mais sauvegarde l'adresse de la prochaine instruction à exécuter (celle qui suit le `CALL`). L'instruction `RET0` sert pour revenir d'un appel de sous-programme (`CALL`). Les deux instructions sont conçues pour fonctionner ensemble.
+
+NOTE: Il est possible de passer autre chose que 0 comme paramètre à `RET`, en effet, `RET` peut être appelé avec des valeurs de `0` à `7`, on discutera de pourquoi plus tard dans le cours.
+
+Pour passer des paramètres, on se contentera pour le moment de passer tout via des registres ou des variables globales. Par conséquent, la récursion est présentement impossible!
```

Comment:
Heheh, I think @R4PaSs wrote this to accentuate that recursion was not the objective of these exercises.